### PR TITLE
Only add root id to OtherPatientIdSequence if it doesn't match with any id in there already.

### PR DESCRIPTION
**Pull request by MarkDechamps**
Only add root id to otherPatientIds if it doesn't already match with any id in there.Otherwise we can get duplicate ids.

### CODE REVIEW

File: dcm4che-core/src/test/java/org/dcm4che3/data/IDWithIssuerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package org.dcm4che3.data;
```
Comment (hczedik) [inline dcm4che-core/src/test/java/org/dcm4che3/data/IDWithIssuerUnitTest.java]: License block (see other files) should be added also here.

File: dcm4che-core/src/test/java/org/dcm4che3/data/IDWithIssuerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.dcm4che3.data.Tag.*;
+import static org.junit.Assert.assertEquals;
+
+public class IDWithIssuerUnitTest {
```
Comment (hczedik) [inline dcm4che-core/src/test/java/org/dcm4che3/data/IDWithIssuerUnitTest.java]: To be consistent with the other names in this package, the unit test class should be called "IDWithIssuerTest".

Please also add a class Javadoc with author (and maybe short explanaition and link to the class under test).

File: dcm4che-core/src/main/java/org/dcm4che3/data/IDWithIssuer.java [hunk 3]
```diff
@@ -149,16 +149,19 @@ public boolean equals(Object obj) {
                     ? other.typeOfPatientID == null
                     : typeOfPatientID.equals(typeOfPatientID)) &&
                 (identifierTypeCode == null
-                    ? other.identifierTypeCode == null
-                    : identifierTypeCode.equals(identifierTypeCode)) &&
+                        ? other.identifierTypeCode == null
```
Comment (hczedik) [inline dcm4che-core/src/main/java/org/dcm4che3/data/IDWithIssuer.java]: Please remove additional formatting.

**Comment by gunterze** [comment]
Resolved by more generic https://github.com/dcm4che/dcm4che/issues/529

**Pull request closed without merging**